v-next/hardhat-ignition/test/migrate/migrate.ts

```diff
@@ -0,0 +1,64 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types/hre";
+
+import { randomUUID } from "node:crypto";
+import { cpSync } from "node:fs";
+import path from "node:path";
+
+import { readJsonFile, remove } from "@nomicfoundation/hardhat-utils/fs";
+import { assert } from "chai";
+
+import taskMigrate from "../../src/internal/tasks/migrate.js";
+
+describe("migrate", function () {
+  const basePath = path.resolve(import.meta.dirname, "..", "fixture-projects");
+  const tmpProjectPath = path.join("tmp", randomUUID());
+  let prevWorkingDir: string;
+
+  // todo: whenever these tests are migrated to node:test,
+  // we should use `useEphemeralFixtureProject` from hardhat-test-utils here instead
+  before(function () {
+    cpSync(
+      path.join(basePath, "migrate-artifacts"),
+      path.join(basePath, tmpProjectPath),
+      {
+        recursive: true,
+        force: true,
+      },
+    );
+
+    prevWorkingDir = process.cwd();
+    process.chdir(path.join(basePath, tmpProjectPath));
+  });
+
+  it("should migrate old artifacts and build-info to the new format", async function () {
```

Comment:
Can you add a test for not changing a v3 deployment? If I followed the logic you have already covered it.